packages/upload-client/src/sharding.js

```diff
@@ -84,3 +88,78 @@ export class ShardingStream extends TransformStream {
     })
   }
 }
+
+/**
+ * @template T
+ * wrap an iterable in a new iterable that will iterate the same items,
+ * but will error if the iterated items aren't sorted according to a comparator
+ * @implements {Iterable<T>}
+ */
+class Sorted {
+  sorted = /** @type {const} */ (true)
+  /**
+   * @param {Iterable<T>} iterable
+   * @param {(a: T, b: T) => number} comparator
+   */
+  constructor(iterable, comparator) {
+    this.iterable = iterable
+    this.comparator = comparator
+  }
+  [Symbol.iterator]() {
+    const { comparator, iterable } = this
+    return function* () {
+      let prev = null
+      for (const cur of iterable) {
+        if (prev && comparator(prev, cur) === 1) {
+          throw Object.assign(
+            new Error(
+              `sequentially iterated items were not sorted as expected`
+            ),
+            { unsorted: [prev, cur], code: 'SORTED_EXPECTATION_UNMET' }
```

Comment:
Again, I'd call this "order".
```suggestion
            { unordered: [prev, cur], code: 'ERR_INVALID_FILE_ORDER' }
```

Reply:
good suggestion, thank you